lib/galaxy/jobs/runners/kubernetes.py

```diff
@@ -749,27 +777,39 @@ def check_watched_item(self, job_state):
                         else:
                             pass
                     else:
+                        log.debug("Job set to running..")
                         job_state.running = True
                         job_state.job_wrapper.change_state(model.Job.states.RUNNING)
                 return job_state
             elif job_persisted_state == model.Job.states.DELETED:
                 # Job has been deleted via stop_job and job has not been deleted,
                 # remove from watched_jobs by returning `None`
+                log.debug("Job is DELETED..")
                 if job_state.job_wrapper.cleanup_job in ("always", "onsuccess"):
                     job_state.job_wrapper.cleanup()
                 return None
             else:
-                return self._handle_job_failure(job, job_state)
+                log.debug(f"Job is failed and not deleted, looking at failure")
+                log.debug(f"Job state before handle job failure: {job_state.job_wrapper.get_state()}")
+                self._handle_job_failure(job, job_state)
+                # changes for resubmission (removed self.mark_as_failed from handle_job_failure)
+                self.work_queue.put((self.mark_as_failed, job_state))
+
+                return None
 
         elif len(jobs.response["items"]) == 0:
             if job_state.job_wrapper.get_job().state == model.Job.states.DELETED:
-                # Job has been deleted via stop_job and job has been deleted,
-                # cleanup and remove from watched_jobs by returning `None`
                 if job_state.job_wrapper.cleanup_job in ("always", "onsuccess"):
                     job_state.job_wrapper.cleanup()
                 return None
```

Comment:
Interactive jobs need here a bit more of attention. Without this part, they keep being shown as running on the history, and you don't get any resulting datasets that are recovered at the end.

Reply:
Has this been resolved?